### PR TITLE
show error message if transaction fee or amount are negative and have…

### PR DESCRIPTION
**Pull request by bp9320**
… more than 2 decimals

### CODE REVIEW

File: src/context/transaction/TransactionState.test.js [hunk 1]
```diff
@@ -53,5 +53,48 @@ describe("Test isPositiveNumber function", () => {
...
+
+  test("Returns false if input is a negative number with more than two decimals", () => {
+    expect(validations.isNumberWithMaxTwoDecimals(-1.234)).toEqual(false);
+  });
+
+  test("Returns true if input is zero", () => {
+    expect(validations.isNumberWithMaxTwoDecimals(0)).toEqual(true);
+  });
+
+  test("Returns false if input is not a number", () => {
```
Comment (zackjp) [inline src/context/transaction/TransactionState.test.js:85]: this description can easily be confused with the `NaN` test. maybe `"...if input a non-Number object"` (or whatever)

File: src/context/transaction/inputValidations.js [hunk 2]
```diff
@@ -13,9 +13,18 @@ const isPositiveNumber = (input) => {
...
-// Validate Transaction amount is two decimals max
 
-// Validate Transaction fee is two decimals max
+// Validate number is two decimals max
+const isNumberWithMaxTwoDecimals = (input) => {
+  if (typeof input !== "number" || isNaN(input)) {
+    return false;
+  }
+  const inputString = input.toString();
+  // const [whole, decimal] = inputString.split(".");
```
Comment (zackjp) [inline src/context/transaction/inputValidations.js:23]: can remove commented out code

File: src/context/transaction/inputValidations.js [hunk 3]
```diff
@@ -13,9 +13,18 @@ const isPositiveNumber = (input) => {
...
+  const decimal = inputString.split(".")[1];
+  return !decimal || (!!decimal && decimal.length <= 2);
```
Comment (zackjp) [inline src/context/transaction/inputValidations.js:25]: it's hard to follow what this means. so, in this context, i believe using variables might help a lot with readability.

File: src/context/transaction/TransactionState.test.js (hunk 1 above)
Comment (bp9320) [reply src/context/transaction/TransactionState.test.js:85]: updated to "if input is not of type 'number'"

File: src/context/transaction/inputValidations.js (hunk 2 above)
Comment (bp9320) [reply src/context/transaction/inputValidations.js:23]: [aaaaaaaand it's gone](https://user-images.githubusercontent.com/32176471/89136983-e578cb00-d4fb-11ea-9b94-f01c0078ba03.png)

File: src/context/transaction/inputValidations.js (hunk 3 above)
Comment (bp9320) [reply src/context/transaction/inputValidations.js:25]: Agree that would take some noodling in the future. New version is much clearer.
